experiment/evaluator.py

```diff
@@ -352,61 +427,80 @@ def do_check_target(self, ai_binary: str, target_path: str) -> Result:
     logger = _Logger(status_path)
 
     # Try building and running the new target.
-    llm_fix_count = 0
-    build_result, run_result = self.builder_runner.build_and_run(
-        generated_oss_fuzz_project, target_path, llm_fix_count)
-    if build_result.succeeded:
-      logger.log(f'Successfully built {target_path} without LLM code fix.')
+
     # TODO: Log build failure.
     # TODO: Log run success/failure.
 
-    # Loop to try and fix the compilation error using the LLM.
-    while not build_result.succeeded and llm_fix_count < LLM_FIX_LIMIT:
+    # Result variables of generation.
+    llm_fix_count = 0
+
+    while True:
```

Comment:
This is a huge change, thanks!
Would you mind helping reduce the size of `while` loop, given it has many tasks? I like the comments you left, maybe we can separate the functionalities into new functions based on them, e.g.:
```python
while not driver_fuzz_err.has_err and llm_fix_count < LLM_FIX_LIMIT:
  cov_pcs, total_pcs, crashes, driver_fuzz_err = build_and_evaluate_fuzz_target()  # Everything belongs to your first comment.
  if driver_fuzz_err.has_err:
    fix_fuzz_target()  # Everything belongs to your 2nd comment. This fixes the fuzz target, the next iteration builds and evaluates it.


# Outside while
if driver_fuzz_err.has_err:
  log(failure info)
  return ...
log(success info)
return ...
```

Please let me know if I missed anything, or if you like to do this together (this will involve a lot code change indeed).
Thanks :)